src/main/java/chungus/util/Pair.java

```diff
@@ -0,0 +1,51 @@
+package chungus.util;
+
+/**
+ * A general pair class.
+ */
+public class Pair<S, T> {
+    private S first;
+    private T second;
```

Comment:
Not essential but consider making these private attributes final since there are no setters.

Reply:
Good idea